apps/aecore/lib/aecore/chain/worker.ex

```diff
@@ -448,19 +446,19 @@ defmodule Aecore.Chain.Worker do
 
     top_chain_state =
       case Persistence.get_all_accounts_chain_states() do
-        chain_states when chain_states == %{} -> state.blocks_data_map[top_hash].chain_state
+        %{} -> state.blocks_data_map[top_hash].chain_state
```

Comment:
AFAIK this doesn't work as intended. `%{}` matches all maps so the first case is always used.

Reply:
(The difference is between how matching and `==` works in Elixir/Erlang. `==` is stricter)

Reply:
Yes, you are correct but see [here](https://stackoverflow.com/questions/27493029/elixir-error-erlang-error-guard-expr?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa) to see that the master code won't work as well apparently

Reply:
Interesting.